packages/datetime/src/dateRangeInput.tsx

```diff
@@ -957,16 +942,16 @@ export class DateRangeInput extends AbstractPureComponent<IDateRangeInputProps,
         const values = this.getStateKeysAndValuesForBoundary(boundary).values;
         const { isInputFocused, hoverString, inputString, selectedValue } = values;
 
-        const boundaryValue = isInputFocused ? this.dateStringToMoment(inputString) : selectedValue;
+        const boundaryValue = isInputFocused ? stringToDate(inputString) : selectedValue;
 
         if (hoverString != null) {
             // don't show an error state while we're hovering over a valid date.
             return false;
-        } else if (isMomentNull(boundaryValue)) {
+        } else if (boundaryValue || boundaryValue == null) {
```

Comment:
Explicit boolean expression only. Also, what's the intended difference in the two checks here?

Reply:
Changed to `boundaryValue == null`